src/services/db/GitFileSystemService.ts

```diff
@@ -203,6 +203,34 @@ export default class GitFileSystemService {
       })
   }
 
+  // Determine if the given branch is present in the local repo copy
+  isLocalBranchPresent(
+    repoName: string,
+    branchName: string
+  ): ResultAsync<boolean, GitFileSystemError> {
+    const efsVolPath = this.getEfsVolPathFromBranch(branchName)
+    return ResultAsync.fromPromise(
+      this.git
+        .cwd({ path: `${efsVolPath}/${repoName}`, root: false })
+        .branchLocal(),
+      (error) => {
+        logger.error(
+          `Unable to get the list of local branches in ${efsVolPath}/${repoName}: ${JSON.stringify(
+            error
+          )}`
+        )
+
+        if (error instanceof GitError) {
+          return new GitFileSystemError(
+            "Unable to get the list of local branches"
+          )
+        }
+
+        return new GitFileSystemError("An unknown error occurred")
+      }
+    ).andThen((result) => okAsync(result.all.includes(branchName)))
```

Comment:
nit: can just use `map` here instead of `andThen`

Reply:
Updated in eb91c65d3632fed7a53946180a9a650e7870dc08.